src/main/java/net/glowstone/util/InventoryUtil.java

```diff
@@ -67,4 +76,46 @@ public static int getRandomSlot(Random random, Inventory inventory, boolean igno
         }
         return notEmpty.get(random.nextInt(notEmpty.size()));
     }
+
+    /**
+     * Inflicts damage to an item. Unbreaking enchantment is applied if present.
+     *
+     * @param player The player holding the item, or null if held by a dispenser
+     * @param holding The item
+     */
+    public static void damageItem(GlowPlayer player, ItemStack holding) {
```

Comment:
What I would do, is make a method with just the `ItemStack`, that returns an `ItemStack`. The ItemStack doesn't get updated in the player's inventory until you use `inventory.setItem(slot, stack)`; and since the slot information isn't inside of the ItemStack object, this current implementation won't work.

Also, in that case, return `InventoryUtil.createEmptyStack()` if the item is to be removed (max damage)

Reply:
Done in 87f6098857730d7a152df87c2d8ed49c15535b69